src/components/CourseInformation/InterestGroupPopup/InterestGroupPopup.test.js

```diff
@@ -0,0 +1,224 @@
+import { act, fireEvent, render, screen } from "@testing-library/react";
+import { Provider } from "react-redux";
+import * as redux from "react-redux";
+import axios from "axios";
+
+import InterestGroupPopup from "./InterestGroupPopup";
+import store from "../../../store/store";
+
+const useSelectorMock = jest.spyOn(redux, "useSelector");
+jest.mock("axios");
+
+let container = null;
+beforeEach(() => {
+  const mockIntersectionObserver = jest.fn();
+  mockIntersectionObserver.mockReturnValue({
+    observe: () => null,
+    unobserve: () => null,
+    disconnect: () => null,
+  });
+  window.IntersectionObserver = mockIntersectionObserver;
+  container = (
+    <div>
+      <Provider store={store}>
+        <InterestGroupPopup courseId={1} />
+      </Provider>
+    </div>
+  );
+});
+afterEach(() => {
+  container = null;
+});
+
+test("does render", async () => {
+  await act(async () => {
+    let lists = [
```

Comment:
consider making a single list at the top of the file so the test cases can be more compact.